cmd/helm/installer/install.go

```diff
@@ -141,6 +142,7 @@ func generateDeployment(opts *Options) *v1beta1.Deployment {
 							},
 							Env: []v1.EnvVar{
 								{Name: "TILLER_NAMESPACE", Value: opts.Namespace},
+								{Name: "TILLER_HISTORY_MAX", Value: fmt.Sprintf("%d", opts.MaxHistory)},
```

Comment:
- [ ] Use [`strconv.Itoa`](https://golang.org/pkg/strconv/#Itoa) instead `fmt.Sprintf`.

Reply:
@seh, are you serious or just trolling? this Sprintf happens once per tiller init/upgrade :)

Reply:
I'm completely serious. I don't review code as a hobby.

My comment was not based on how often this statement executes, nor whether or not it _ever_ executes. I am able to see from its context and its intention that it is using too powerful a tool to do too little, and that the standard library already anticipated this need.

By analogy, if you have a dull pencil that you want to sharpen, `strconv.Itoa` is a pencil sharpener: purpose-built to accept this lone input, with a predictable technique for producing the promised output. By contrast, `fmt.Sprintf` is the wood shop, full of machines and tools that one could adapt for use in sharpening a pencil, but with more labor, more complication, and more chance of getting it wrong.

This is not just a matter of execution time cost. It is a matter of unnecessary flexibility that _also_ happens to bears an execution time cost.

Do you think that `fmt.Sprintf` is a better choice here, or is your point that you don't care how the code works, as long as it doesn't run frequently?

Reply:
you realize you spend more time on this comment, than all users around the world would saved should you replace `fmt.Sprintf` with `strconf.Itoa` in this particular place? :)

`fmt.Sprintf` is deterministic and pure, so matches your criteria of "with a predictable technique for producing the promised output", besides that flexibility argument wasn't anywhere in your original comment, it was all about speed.

I don't think `fmt.Sprintf` is a better choice, I think it is equal choice. I don't care how code works as long as it produces correct result. If speed is a problem, than it is a different conversation.

Reply:
I am willing to spend as much time as necessary writing and reading in order to reach accord on matters like these, regardless of the real-world execution time of a particular statement in the code. This is teamwork.

My original comment was a prescribed replacement, and my elaboration—upon request—was about how much code is involved in each respective implementation. You can interpret its focus as you will.

I think we understand each other here, even if we have different preferences. I just wanted to clarify that my prescription was not capricious, nor written in jest; it comes from considered way of thinking. We don't have to agree on this, but code review is a chance to suggest improvements, debate choices, correct mistakes, and learn about each other. In that, we've been successful.